### PR TITLE
Fixed doomsday device spamming.

### PR DESCRIPTION
**Pull request by xxalpha**
Fixes https://github.com/tgstation/-tg-station/issues/15170

### CODE REVIEW

**Comment by lzimann** [comment]
thx

**Comment by KorPhaeron** [comment]
Why didn't qdel() stop it from processing?

**Comment by xxalpha** [comment]
The problem isn't qdel(), the problem is it has no xyz when inside the AI.

**Comment by KorPhaeron** [comment]
@xxalpha 

No I realize that is an issue as well, but it is meant to qdel a few lines down.

I indented it twice instead though, could you please fix that?

File: code/game/gamemodes/malfunction/Malf_Modules.dm [hunk 1]
```diff
@@ -59,18 +59,18 @@
 	var/timer = 450
 
 /obj/machinery/doomsday_device/process()
-	if(z != ZLEVEL_STATION)
-		minor_announce("DOOMSDAY DEVICE OUT OF RANGE OF STATION, ABORTING", "ERROR ER0RR $R0RRO$!R41.%%!!(%$^^__+ @#F0E4", 1)
+	var/turf/T = get_turf(src)
+	if(T && T.z != ZLEVEL_STATION)
```
Comment (KorPhaeron) [inline code/game/gamemodes/malfunction/Malf_Modules.dm]: Probably does not matter either way, but 

if(!T || T.z != ZLEVEL_STATION)

would catch edgecases like ending up in nullspace

Comment (xxalpha) [inline code/game/gamemodes/malfunction/Malf_Modules.dm]: T can exist in nullspace, so ``!T`` wouldn't do much regarding that.

**Pull request merged**